### PR TITLE
convert cronjob to chart-job

### PR DESCRIPTION
**Pull request by NickAzureDevops**
### Jira link (https://tools.hmcts.net/jira/browse/DTSPO-18508)

convert existing cronjob to chart-job 

### Checklist

- [x] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)
- [ ] Does this PR introduce a breaking change

## 🤖AEP PR SUMMARY🤖

_I'm a bot that generates AI summaries of pull requests, see [AEP](https://kainossoftwareltd.github.io/ai-enhanced-platform/) for more details_

- Updated `check-acr-sync.yaml` to use `HelmRelease` from `CronJob` and adjust configuration accordingly.
- Added a new file `helm-base-versions-slack.yaml` for Slack notifications related to base versions.
- Updated `kustomization.yaml` to remove patches and include `helm-base-versions-slack.yaml`.
- Updated `helm-base-versions.yaml` to use `HelmRelease` from `CronJob` and adjust configuration accordingly.
- Excluded `check-acr-sync.yaml` from the `flux-v2-image-automation.sh` script.

### CODE REVIEW

File: apps/monitoring/helm-base-versions/helm-base-versions.yaml [hunk 1]
```diff
@@ -1,36 +1,53 @@
...
+      - key: dedicated
+        effect: NoSchedule
+        operator: Equal
+        value: jobs
+    jobTemplate:
+      spec:
+        template:
+          metadata:
+            labels:
+              azure.workload.identity/use: "true"
```
Comment (MartyFox) [inline apps/monitoring/helm-base-versions/helm-base-versions.yaml:53]: Shouldnt need this if using the chart-job helm chart, it has this as an option/value input and is defaulted to true
https://github.com/hmcts/chart-job/blob/master/job/values.yaml#L35

File: apps/monitoring/helm-base-versions/helm-base-versions.yaml [hunk 2]
```diff
@@ -1,36 +1,53 @@
...
+    backoffLimit: 3
+    restartPolicy: Never
+    runAsRoot: true
+    kind: CronJob
+    image: hmctspublic.azurecr.io/helm-base-versions:prod-0bf5a71-20240814143926Z #{"$imagepolicy": "flux-system:helm-base-versions"}
+    imagePullPolicy: Always
+    environment:
+      CLUSTER_NAME: "cft-${CLUSTER_FULL_NAME}-aks"
+      SLACK_WEBHOOK:
+        valueFrom:
```
Comment (MartyFox) [inline apps/monitoring/helm-base-versions/helm-base-versions.yaml:35]: Environment variables from Key Vault should be done via the KeyVaults values
```yaml
    keyVaults:
     key-vault-name:
        excludeEnvironmentSuffix: true
        secrets:
          - name: secretName
            alias: ENV_VAR_NAME
```
https://github.com/hmcts/chart-job?tab=readme-ov-file#adding-azure-key-vault-secrets

**Comment by NickAzureDevops** [comment]
no longer need to use kv as it is picking up from aks secrets

File: apps/monitoring/helm-base-versions-slack/helm-base-versions-slack.yaml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    spec:
+      chart: job
+      sourceRef:
+        kind: GitRepository
+        name: chart-job
+        namespace: flux-system
+      interval: 1m
+  interval: 5m
+  values:
+    schedule: "0 9 * * 1-5"
```
Comment (MartyFox) [inline apps/monitoring/helm-base-versions-slack/helm-base-versions-slack.yaml:21]: This doesnt match the numbers from the patch that in place `"00 10 * * 1-5"`

File: apps/monitoring/acr-sync/check-acr-sync.yaml [hunk 4]
```diff
@@ -1,50 +1,61 @@
...
+        namespace: flux-system
+      interval: 1m
+  interval: 5m
+  values:
+    cronJob:
+      schedule: "*/15 * * * *"
+      concurrencyPolicy: "Forbid"
+      failedJobsHistoryLimit: 3
+      startingDeadlineSeconds: 300
+      jobTemplate:
```
Comment (MartyFox) [inline apps/monitoring/acr-sync/check-acr-sync.yaml:24]: Should `jobTemplate` be in this file if its converting to a helmRelease chart-job type?

File: apps/monitoring/helm-base-versions-slack/kustomization.yaml [hunk 5]
```diff
@@ -3,17 +3,4 @@ kind: Kustomization
 nameSuffix: "-slack"
 resources:
   - ../helm-base-versions/helm-base-versions.yaml
```
Comment (MartyFox) [inline apps/monitoring/helm-base-versions-slack/kustomization.yaml:5]: I dont think you need this line now, this was here and then being patched, you've created a standalone file for the slack side

Comment (NickAzureDevops) [reply apps/monitoring/helm-base-versions-slack/kustomization.yaml:5]: ooh i see that make sense as it was a patched from the other file

**Review by MartyFox — APPROVED** [review]

**Pull request merged**